_summaries/sukhbaatar2015end-to-end_2.md

```diff
@@ -1,19 +1,42 @@
 ---
 layout: summary
 title: Summary
-paper: {{paper_tag}}
+paper: {{sukhbaatar2015end-to-end}}
 # Please fill out info below
-author: # Your GitHub id
-score: # How did you like this paper 0(dislike) to 10(love)
+author: mmcinnestaylor # Your GitHub id
+score: 8 # How did you like this paper 0(dislike) to 10(love)
 ---
 
-TODO: Summarize the paper:
-* What is the core idea?
-* How is it realized (technically)?
-* How well does the paper perform?
-* What interesting variants are explored?
+## Core Idea
+* End-To-End Memory Networks are a type of Memory Network with a possibly large external memory which is are trained end-to-end.
+* End-to-end training requires less supervision, thus improving applicability in real world settings.
+
+## Technical Implementation
+![Architecture](sukhbaatar2015end-to-end_2_b.png) 
+* The model accepts a discrete set of inputs (a story) along with a query, and outputs an answer.  
+![Q&A](sukhbaatar2015end-to-end_2_a.png)  
+* The inputs {x} and query q are transformed using embedding matrices A and B, respectively.  
+$$m_i = \text{Embed}_A(x_i)\qquad u = \text{Embed}_B(q)$$
+* The the most relevant embedded story input is matched with the embedded query by taking the inner product of each input with the query followed by a softmax operation, thus producing a probability vector.  
+$$p_i = \text{softmax}(u^{\text{T}}m_i)$$  
+* The unembedded story inputs are then embedded once again using embedding matrix C.  
+$$c_i = \text{Embed}_C(x_i)$$  
+* The response vector is then calculated by summing the inputs in {c} and weighting each with the probability vector from the previous step.  
+$$o = \sum_ip_ic_i$$  
+* The final predicton, or answer to the question, is then calculated by summing the response vector with the initially embedded query, passing it through a weight matrix, and then applying a softmax operation.  
+$$\hat{a} = \text{softmax}(W(o + u))$$  
+* During training, utilizing stochastic gradient descent, the matrices A, B, C, and W are learned by minimizing the standard cross-entropy loss between the predicted label, and the correct label.  
+* The architecture can be...  
+## Variants 
+* Along with the baseline model, the authors experimented with deepening the architecture by adding additional layers, which they term 'hops'.  
```

Comment:
Might want to dig into what "hops" are (additional run-throughs of the model almost like RNNs).  Addtional layers may be confusing since its not really a "larger" network, but just ran more.